addons/binding/org.openhab.binding.connectsdk/src/main/java/org/openhab/binding/connectsdk/handler/ConnectSDKHandler.java

```diff
@@ -0,0 +1,235 @@
+import com.connectsdk.service.DeviceService;
+import com.connectsdk.service.DeviceService.PairingType;
+import com.connectsdk.service.command.ServiceCommandError;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * The {@link ConnectSDKHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Sebastian Prehn - Initial contribution
+ */
+public class ConnectSDKHandler extends BaseThingHandler implements ConnectableDeviceListener, DiscoveryManagerListener {
+
+    private Logger logger = LoggerFactory.getLogger(ConnectSDKHandler.class);
+    private DiscoveryManager discoveryManager;
+
+    // ChannelID to CommandHandler Map
+    private final Map<String, ChannelHandler> channelHandlers = ImmutableMap.<String, ChannelHandler>builder()
+            .put(CHANNEL_VOLUME, new VolumeControlVolume()).put(CHANNEL_VOLUME_UP, new VolumeControlUp())
+            .put(CHANNEL_VOLUME_DOWN, new VolumeControlDown()).put(CHANNEL_POWER, new PowerControlPower())
+            .put(CHANNEL_MUTE, new VolumeControlMute()).put(CHANNEL_CHANNEL, new TVControlChannel())
+            .put(CHANNEL_CHANNEL_UP, new TVControlUp()).put(CHANNEL_CHANNEL_DOWN, new TVControlDown())
+            .put(CHANNEL_CHANNEL_NAME, new TVControlChannelName()).put(CHANNEL_APP_LAUCHER, new LauncherApplication())
+            .put(CHANNEL_MEDIA_FORWARD, new MediaControlForward()).put(CHANNEL_MEDIA_PAUSE, new MediaControlPause())
+            .put(CHANNEL_MEDIA_PLAY, new MediaControlPlay()).put(CHANNEL_MEDIA_REWIND, new MediaControlRewind())
+            .put(CHANNEL_MEDIA_STOP, new MediaControlStop()).put(CHANNEL_MEDIA_STATE, new MediaControlPlayState())
+            .put(CHANNEL_TOAST, new ToastControlToast()).build();
+
+    public ConnectSDKHandler(Thing thing, DiscoveryManager discoveryManager) {
+        super(thing);
+        this.discoveryManager = discoveryManager;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("internalReceiveCommand({},{}) is called", channelUID, command);
+        ChannelHandler handler = channelHandlers.get(channelUID.getId());
+        final ConnectableDevice d = getDevice();
+        if (d == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    String.format("%s not found under connect sdk devices", getThing().getUID()));
+            return;
+        }
+        handler.onReceiveCommand(d, command);
+    }
+
+    private ConnectableDevice getDevice() {
+        String ip = this.getThing().getProperties().get(PROPERTY_IP_ADDRESS);
+        return this.discoveryManager.getCompatibleDevices().get(ip);
+    }
+
+    @Override
+    public void initialize() {
+        this.discoveryManager.addListener(this);
+
+        ConnectableDevice d = getDevice();
+        if (d == null) {// If TV is off getDevice() will return null
+            updateStatus(ThingStatus.OFFLINE);
```

Comment:
You can use ThingStatusDetail.NONE and provide message “TV is off”. My comment is about this message - it will be displayed next to the thing status and make it clear it is not because of some issues happening.

Reply:
done